source/fileio/write_network.py

```diff
@@ -225,11 +233,28 @@ def write(self, flownetwork):
         if flownetwork.ht is not None:
             graph.es["ht"] = flownetwork.ht
 
+        if flownetwork.hd is not None:
+            graph.es["hd"] = flownetwork.hd
+
         graph.vs["xyz"] = flownetwork.xyz.tolist()
 
         if flownetwork.pressure is not None:
             graph.vs["pressure"] = flownetwork.pressure
 
+        # Edges
+        not_convergence_vessel = np.zeros(flownetwork.nr_of_es)
+        not_convergence_vessel[flownetwork.vessel_general] = 1
+        if flownetwork.vessel_general is not None:
+            graph.es["not_convergence_vessel"] = not_convergence_vessel
+
+        # Vertex
+        not_convergence_node = np.zeros(flownetwork.nr_of_vs)
+        not_convergence_node[flownetwork.indices_over] = 1
+        not_convergence_node_blue = np.zeros(flownetwork.nr_of_vs)
+        not_convergence_node_blue[flownetwork.indices_over_blue] = 1
+        if flownetwork.vessel_general is not None:
+            graph.vs["not_convergence_node"] = not_convergence_node
```

Comment:
Same here! I guess to avoid errors when we run different test cases, 251-254 lines should be inside this if condition.